sonarcloud.sh

```diff
@@ -0,0 +1,98 @@
+#! /bin/bash
+
+##
+# Copyright Ladislas de Toldi 2018
+# Copyright IBM Corporation 2016,2017,2018
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##
+
+if [[ $TRAVIS_BRANCH != "master" && $TRAVIS_BRANCH != "develop" && $TRAVIS_EVENT_TYPE != "cron" ]]; then
+  echo "Not master, develop or cron build. Skipping code coverage generation."
+  exit 0
+fi
+
+echo ">> Starting Sonar Cloud code coverage analysis..."
+uname -a
+
+SDK=macosx
+DERIVED_DATA=.build/scanner
+xcodebuild -version
+xcodebuild -version -sdk $SDK
+if [[ $? != 0 ]]; then
+  exit 1
+fi
+
+# Determine if there is a custom command for generating xcode project
+source ./Package-Builder/generate-xcodeproj.sh
+
+# Determine if there is a custom command for xcode build (code coverage tests)
+if [ -e ${projectFolder}/.swift-codecov ]; then
+  XCODE_BUILD_CMD=$(cat ${projectFolder}/.swift-codecov)
+else
+  PROJECT="${PROJ_OUTPUT##*/}"
+  SCHEME=$(xcodebuild -list -project $PROJECT | grep --after-context=1 '^\s*Schemes:' | tail -n 1 | xargs)
+  XCODE_BUILD_CMD="xcodebuild -project $PROJECT -scheme $SCHEME -derivedDataPath $DERIVED_DATA -enableCodeCoverage YES clean build test CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO"
+fi
+
+echo ">> Running: $XCODE_BUILD_CMD"
+eval "$XCODE_BUILD_CMD"
+if [[ $? != 0 ]]; then
+  echo ">> Error running: $XCODE_BUILD_CMD"
+  exit 1
+fi
+
+BASH_URL="https://raw.githubusercontent.com/SonarSource/sonar-scanning-examples/master/swift-coverage/swift-coverage-example/xccov-to-sonarqube-generic.sh"
+BASH_BASE="bash <(curl -s $BASH_URL)"
```

Comment:
Hmm is it a good idea to execute arbitrary code, probably not?

Reply:
Nope, you're right, I've added the file into the repo and linked to it.

Reply:
Now it's in the repo we don't need to `curl` it do we?

Reply:
Just fixed it!